### PR TITLE
Feat/add gpt oss sequence classification

### LINKED ISSUE #40018: need GptOssForSequenceClassification
### Feature request

need GptOssForSequenceClassification

### Motivation

for text classificition

### Your contribution

nothing

### PR DESCRIPTION
**Pull request by robin-ede**
# What does this PR do?

This PR implements `GptOssForSequenceClassification` for text classification tasks.

**Key Changes:**
- ✅ **New Model Class**: Added `GptOssForSequenceClassification` inheriting from `GenericForSequenceClassification` and `GptOssPreTrainedModel`
- ✅ **Consistent Implementation**: Implemented in both `modeling_gpt_oss.py` (generated) and `modular_gpt_oss.py` (source) files
- ✅ **Auto-Model Integration**: Added to `modeling_auto.py` registry for automatic model discovery
- ✅ **Test Infrastructure**: Added `sequence_classification_class` to test configuration and enabled all sequence classification tests
- ✅ **Pipeline Support**: Configured for `text-classification` and `zero-shot` pipelines

**Implementation Details:**
- Follows HuggingFace's standard pattern used by other MoE models (Qwen2MoE, Mixtral)
- Uses `GenericForSequenceClassification` mixin for automatic classification head and forward pass logic
- Maintains compatibility with existing GptOss architecture and weight initialization
- Supports both single-label and multi-label classification scenarios

**Testing:**
- All sequence classification tests pass (3/3): basic, single-label, and multi-label variants
- Model instantiation and forward pass work correctly with proper logits shape `[batch_size, num_labels]`
- Integration with HuggingFace pipelines confirmed

Fixes #40018

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
      **Link:** https://github.com/huggingface/transformers/issues/40018
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
      **Yes:** Added `GptOssForSequenceClassification` to `docs/source/en/model_doc/gpt_oss.md` for auto-generated documentation
- [x] Did you write any new necessary tests?
      **Yes:** Added `sequence_classification_class = GptOssForSequenceClassification` to test configuration, enabling all sequence classification tests

## Who can review?

@ArthurZucker - This adds a new text model class following standard patterns for sequence classification

### CODE REVIEW

File: src/transformers/models/gpt_oss/configuration_gpt_oss.py [hunk 1]
```diff
@@ -69,6 +69,7 @@ def __init__(
         output_router_logits=False,
         use_cache=True,
         layer_types=None,
+        num_labels=2,
```
Comment (ArthurZucker) [inline src/transformers/models/gpt_oss/configuration_gpt_oss.py]: we should not have to add this

File: src/transformers/models/gpt_oss/modeling_gpt_oss.py [hunk 2]
```diff
@@ -700,4 +700,8 @@ def forward(
         )
 
 
-__all__ = ["GptOssForCausalLM", "GptOssModel", "GptOssPreTrainedModel"]
+class GptOssForSequenceClassification(GenericForSequenceClassification, GptOssPreTrainedModel):
+    pass
```
Comment (ArthurZucker) [inline src/transformers/models/gpt_oss/modeling_gpt_oss.py:707]: nice!

**Review by ArthurZucker — COMMENTED** [review]
Great except for the num labels!

**Comment by robin-ede** [comment]
@ArthurZucker all done!

**Comment by ArthurZucker** [comment]
Hey sorry by the time I got back I merged another PR that did the same! Super sorry!

**Comment by robin-ede** [comment]
No worries!
